exaudfclient/base/javacontainer/test/cpp/javacontainer_test.cc

```diff
@@ -0,0 +1,58 @@
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_classpath, "/exaudf/javacontainer/libexaudf.jar:javacontainer/test/test.jar");
+    const std::vector<std::string> expectedJarPaths = {"javacontainer/test/test.jar"};
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_jarPaths, expectedJarPaths);
+    EXPECT_FALSE(vm.getJavaVMInternalStatus().m_needsCompilation);
+    const std::vector<std::string> expectedJVMOptions = { "-Dexasol.scriptclass=com.exasol.udf_profiling.UdfProfiler",
+                                                            "-Xms128m", "-Xmx128m", "-Xss512k",
+                                                            "-XX:ErrorFile=/tmp/hs_err_pid%p.log",
+                                                            "-Djava.class.path=/exaudf/javacontainer/libexaudf.jar:javacontainer/test/test.jar",
+                                                            "-XX:+UseSerialGC" };
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_jvmOptions, expectedJVMOptions);
+}
+
+
+TEST(JavaContainer, basic_inline) {
+    const std::string script_code = "import java.time.LocalDateTime;"
+                                    "import java.time.ZoneOffset;"
+                                    "import java.time.format.DateTimeFormatter;"
+                                    "class SIMPLE {"
+                                    "static void main(String[] args) {"
+                                    "int i = 0;"
+                                    "}"
+                                    "static int run(ExaMetadata exa, ExaIterator ctx) {"
+                                    "return 0;"
+                                    "}"
+                                    "}";
+    JavaVMTest vm(script_code);
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_exaJavaPath, "/exaudf/javacontainer");
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_localClasspath, "/tmp");
+    const std::string expected_script_code = std::string("package com.exasol;\r\n") + script_code;
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_scriptCode, expected_script_code);
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_exaJarPath, "/exaudf/javacontainer/libexaudf.jar");
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_classpath, "/tmp:/exaudf/javacontainer/libexaudf.jar");
+    const std::vector<std::string> expectedJarPaths;
+    EXPECT_TRUE(vm.getJavaVMInternalStatus().m_jarPaths.empty());
+    EXPECT_TRUE(vm.getJavaVMInternalStatus().m_needsCompilation);
+    const std::vector<std::string> expectedJVMOptions = {   "-Xms128m", "-Xmx128m", "-Xss512k",
+                                                            "-XX:ErrorFile=/tmp/hs_err_pid%p.log",
+                                                            "-Djava.class.path=/tmp:/exaudf/javacontainer/libexaudf.jar",
+                                                            "-XX:+UseSerialGC" };
+    EXPECT_EQ(vm.getJavaVMInternalStatus().m_jvmOptions, expectedJVMOptions);
+}
+
```

Comment:
Do we need a test for script and jar together?

Reply:
Done